packages/ckb-sdk-core/src/index.ts

```diff
@@ -420,6 +420,49 @@ class CKB {
     }
   }
 
+  public calculateDaoMaximumWithdraw = async (
+      outPoint: CKBComponents.OutPoint,
+      withdrawBlockHash: string
+  ): Promise<string> => {
+    const depositTx = await this.rpc.getTransaction(outPoint.txHash)
+    if (depositTx.txStatus.status !== 'committed') throw new Error('Transaction is not committed yet')
+    const depositCell = depositTx.transaction.outputs[+outPoint.index]
+    const depositBlockHash = depositTx.txStatus.blockHash
+    const depositHeader = await this.rpc.getHeader(depositBlockHash)
+    const depositAr = this.#extracDaoData(depositHeader.dao).ar
+    const withDrawHeader = await this.rpc.getHeader(withdrawBlockHash)
+    const withdrawAr = this.#extracDaoData(withDrawHeader.dao).ar
+    const { JSBI } = this.utils
+    const occupiedCapacity = JSBI.asUintN(64, JSBI.BigInt(10200000000));
```

Comment:
The `occupied capacity` is dynamic according to 
![image](https://user-images.githubusercontent.com/7271329/155956000-4dcdd8c9-8fea-4ae2-8c2e-3e671676a51e.png)
`10200000000` is just for the default lock in the example.

Reply:
According to [ckb-explorer](https://github.com/nervosnetwork/ckb-explorer/blob/develop/app/utils/dao_compensation_calculator.rb#L25), the `occupied capacity` is determined by `withdraw cell`, when the `withdrawal` doesn't occur, we can assume the `cell` won't change and use `deposit cell` instead of `withdraw cell`.